consensus/spos/bls/subroundSignature.go

```diff
@@ -379,58 +383,84 @@ func (sr *subroundSignature) remainingTime() time.Duration {
 }
 
 func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
-	isCurrentNodeMultiKeyLeader := sr.IsMultiKeyLeaderInCurrentRound()
-	isFlagActive := sr.EnableEpochsHandler().IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, sr.Header.GetEpoch())
 
-	numMultiKeysSignaturesSent := 0
+	numMultiKeysSignaturesSent := int32(0)
+	sentSigForAllKeys := atomicCore.Flag{}
+	sentSigForAllKeys.SetValue(true)
+
+	wg := sync.WaitGroup{}
+
 	for idx, pk := range sr.ConsensusGroup() {
+		wg.Add(1)
```

Comment:
the add should be done at line 405 then you don't need the wg.Done at lines 397 & line 402